modules/haloRtdProvider.js

```diff
@@ -88,20 +142,20 @@ function getSegments(reqBidsConfigObj, onDone, config, userConsent) {
  * @param {Object} userConsent
  * @param {Object} userIds
  */
-function getSegmentsAsync(adUnits, onDone, config, userConsent, userIds) {
-  let reqParams = {}
+export function getSegmentsAsync(adUnits, onDone, config, userConsent, userIds) {
+  let reqParams = {};
   if (typeof config == 'object' && config != null && Object.keys(config).length > 0) {
-    reqParams = config.params
+    reqParams = config.params.requestParams;
```

Comment:
may want to check this exists first and use the **init** function of your submodule to check that you have the required params and send a logError if any are missing